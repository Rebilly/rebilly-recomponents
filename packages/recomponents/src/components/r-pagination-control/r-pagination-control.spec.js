import {shallowMount} from '@vue/test-utils';
import {renderToString} from '@vue/server-test-utils';
import RPaginationControl from './r-pagination-control.vue';


describe('r-pagination-control.vue', () => {
    it('should render Wrapper and match snapshot', () => {
        const wrapper = shallowMount(RPaginationControl);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render via SSR and match snapshot', async () => {
        const wrapper = await renderToString(RPaginationControl);
        expect(wrapper).toMatchSnapshot();
    });

    it('pages count correctly when total/limit < 1.5', () => {
        const wrapper = shallowMount(RPaginationControl, {
            propsData: {
                total: 5,
                limit: 4,
            },
        });

        expect(wrapper.vm.totalPages).toEqual(2);
    });

    it('pages count correctly when total/limit > 1.5', () => {
        const wrapper = shallowMount(RPaginationControl, {
            propsData: {
                total: 5,
                limit: 2,
            },
        });

        expect(wrapper.vm.totalPages).toEqual(3);
    });

    it('hasNext property correctly', () => {
        const wrapper = shallowMount(RPaginationControl, {
            propsData: {
                total: 5,
                limit: 2,
            },
        });

        expect(wrapper.vm.canMoveForward).toEqual(true);
        wrapper.vm.navigate(3);
        expect(wrapper.vm.canMoveForward).toEqual(false);
    });

    it('hasPrevious property correctly', () => {
        const wrapper = shallowMount(RPaginationControl, {
            propsData: {
                total: 5,
                limit: 2,
            },
        });

        expect(wrapper.vm.canMoveBackward).toEqual(false);
        wrapper.vm.navigate(3);
        expect(wrapper.vm.canMoveBackward).toEqual(true);
    });

    it('items property correctly', () => {
        const wrapper = shallowMount(RPaginationControl, {
            propsData: {
                total: 100,
                limit: 10,
                totalVisible: 5,
            },
        });

        const getVisible = items => items.map(({id}) => id);
        expect(getVisible(wrapper.vm.pages)).toEqual([1, 2, 3, 4, 5]);
        wrapper.vm.navigate(5);
        expect(getVisible(wrapper.vm.pages)).toEqual([3, 4, 5, 6, 7]);
        wrapper.vm.navigate(10);
        expect(getVisible(wrapper.vm.pages)).toEqual([6, 7, 8, 9, 10]);
    });
});
