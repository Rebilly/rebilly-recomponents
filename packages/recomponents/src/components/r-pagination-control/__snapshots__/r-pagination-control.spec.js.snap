// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`r-pagination-control.vue should render Wrapper and match snapshot 1`] = `
<div>
  <div class="r-pagination-control">
    <div class="r-pagination-control-buttons-wrapper">
      <r-button-stub size="regular" type="default" disabled="true" class="r-pagination-control-button">
        <r-icon-stub icon="arrow-left"></r-icon-stub>
        Prev
      </r-button-stub>
      <r-button-stub size="regular" type="default" disabled="true" class="r-pagination-control-button">
        Next
        <r-icon-stub icon="arrow-right"></r-icon-stub>
      </r-button-stub>
    </div>
  </div>
</div>
`;

exports[`r-pagination-control.vue should render via SSR and match snapshot 1`] = `
<div>
  <div class="r-pagination-control">
    <div class="r-pagination-control-buttons-wrapper"><button role="button" disabled="disabled" title="Action is disabled" class="r-button r-pagination-control-button r-button-size-regular r-button-type-default">
        <!---->
        <div class="capitalize-first-letter"><svg class="r-icon r-icon-20">
            <use xlink:href="#icon-arrow-left" xmlns:xlink="http://www.w3.org/1999/xlink"></use>
          </svg>
          Prev
        </div>
      </button> <button role="button" disabled="disabled" title="Action is disabled" class="r-button r-pagination-control-button r-button-size-regular r-button-type-default">
        <!---->
        <div class="capitalize-first-letter">
          Next
          <svg class="r-icon r-icon-20">
            <use xlink:href="#icon-arrow-right" xmlns:xlink="http://www.w3.org/1999/xlink"></use>
          </svg></div>
      </button></div>
  </div>
</div>
`;
